### PR TITLE
externals: Switch to newer cryptopp-cmake.

### PR DESCRIPTION
**Pull request by Steveice10**
Follow-up to https://github.com/citra-emu/citra/pull/6240.

Replaces Citra's customized version of https://github.com/noloader/cryptopp-cmake with its successor, https://github.com/abdes/cryptopp-cmake. The newer cryptopp-cmake is more modern and takes into account variables like CMAKE_OSX_ARCHITECTURES, so there is no longer a need to explicitly disable ASM for universal builds. Additionally, there are new flags to apply a lot of customizations previously edited into Citra's copy.

This PR also updates the CMake version for the Android build for compatibility with the new submodule.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6242)

### CODE REVIEW

File: src/android/app/build.gradle [hunk 1]
```diff
@@ -96,7 +96,7 @@ android {
 
     externalNativeBuild {
         cmake {
-            version "3.18.1"
+            version "3.25.1"
```
Comment (SachinVin) [inline src/android/app/build.gradle:99]:
```suggestion
            version "3.22.1"
```
Preferably stick to versions available in the Android SDK, so gradle can automatically fetch them when they are not available

**Comment by Steveice10** [comment]
Looking into why cryptopp include is failing after switching Android to CMake 3.22.1

File: .gitmodules [hunk 2]
```diff
@@ -58,3 +55,6 @@
 [submodule "sdl2"]
 	path = externals/sdl2/SDL
 	url = https://github.com/libsdl-org/SDL
+[submodule "externals/cryptopp-cmake"]
```
Comment (FearlessTobi) [inline .gitmodules:58]:
```suggestion
[submodule "cryptopp-cmake"]
```

**Comment by Steveice10** [comment]
There appears to be some issue with the cmake_path PARENT_PATH command in the Android SDK's build of CMake 3.22.1. cryptopp-cmake uses it to get the parent path of the cryptopp include directory and add that to the include paths as well, so you can do includes prefixed with ```cryptopp/```. However on that specific CMake PARENT_PATH is yielding the exact same path, so includes of that type fail.

I've implemented a hack for Android for the time being until the SDK has a CMake build without this issue, which fixes the Android build. I've also addressed current review comments.

**Pull request merged**